controllers/ratelimitpolicy_controller.go

```diff
@@ -201,7 +201,7 @@ func (r *RateLimitPolicyReconciler) deleteResources(ctx context.Context, rlp *ku
 		return err
 	}
 
-	if err := r.deleteLimits(ctx, rlp); err != nil && !apierrors.IsNotFound(err) {
+	if err := r.reconcileLimits(ctx, rlp); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
> This line runs before updating the gateway back references to the policies. When cleaning up for a RLP just deleted, we need to make sure the limits are reconciled without the reference to such policy. This was the main difference between reconcileLimits and deleteLimits.

I've done this as there was an issue with the `AffectedPolicyMap` when a policy is deleted since it will never be removed from it for deleted policies. The particular case was GW policy with overrides and deleted and route policy was still marked as affected / overridden because of it which was needed at:
https://github.com/KevFan/kuadrant-operator/blob/ba9922086074752965ba412809a7edb1ae58faf6/controllers/ratelimitpolicy_limits.go#L152-L154

Which was why I've opted to skip the deleted RLP limits instead at:
https://github.com/KevFan/kuadrant-operator/blob/ba9922086074752965ba412809a7edb1ae58faf6/controllers/ratelimitpolicy_limits.go#L118-L122 

> It looks like now we want the deleted RLP to be treated as if it was still affecting the gateway. Furthermore, if the API server has finished deleting the resource already, then this line shall fail, no? https://github.com/KevFan/kuadrant-operator/blob/ba9922086074752965ba412809a7edb1ae58faf6/controllers/ratelimitpolicy_limits.go#L110

Yes, true, there's the assumption that for the first reconcile, this should pass as the resource is not fully deleted yet, and once this completes the references is removed from the GW, so any secondary reconciles will not fail at this stage. I guess this to me, significifes the issue with how each RLP reconciled must get all the RLPs in order to consolidate into one Limitador CR, so using the `affectedPolicy` map for determining enforced status is not ideal